### DISCUSSION TITLE
Finding holes in a polygon coverage

### DISCUSSION THREAD

**Discussion by micycle1** [locationtech/jts#986] [Q&A] [answered]
What is the best (fastest) approach to detect holes in a polygon coverage (i.e holes in a triangular mesh)?

One approach I've found is creating a polygon from the coverage (using `CoverageUnion.union`) and then counting its interior rings, but is there a better way?

**Comment by dr-jts** [comment, marked as answer]
`CoverageUnion` is pretty efficient, so if you want an easy way to detect holes that's the best thing to use.  Theoretically you might be able to do it a bit faster with a custom algorithm, but that's a lot of work for a small performance gain.
